src/main/java/frc/robot/subsystems/shooter/ShooterFlywheel.java

```diff
@@ -0,0 +1,84 @@
+package frc.robot.subsystems.shooter;
+
+import org.littletonrobotics.junction.Logger;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.SimpleMotorFeedforward;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+
+public class ShooterFlywheel extends SubsystemBase
+{
+    private ShooterFlywheelIO                       _flywheelIO;
+    private final ShooterFlywheelIOInputsAutoLogged _inputs                = new ShooterFlywheelIOInputsAutoLogged();
+    private SimpleMotorFeedforward                  _flywheelFeedforward;
+    private PIDController                           _upperFlywheelFeedback;
+    private PIDController                           _lowerFlywheelFeedback;
+    private Double                                  _upperVelocitySetpoint = null;
+    private Double                                  _lowerVelocitySetpoint = null;
+    private double                                  _maxFlywheelSpeed      = Constants.ShooterFlywheel.MAX_FLYWHEEL_SPEED;
+
+    public ShooterFlywheel(ShooterFlywheelIO flywheelIO)
+    {
+        _flywheelIO = flywheelIO;
+
+        _flywheelFeedforward   = new SimpleMotorFeedforward(0.01, 0.01);
+        _upperFlywheelFeedback = new PIDController(0.01, 0, 0);// FIXME: Set values, calibrate
+        _lowerFlywheelFeedback = new PIDController(0.01, 0, 0);
+    }
+
+    @Override
+    public void periodic()
+    {
+        _flywheelIO.updateInputs(_inputs);
+        Logger.processInputs("Shooter/Flywheel", _inputs);
+
+        if (_upperVelocitySetpoint != null)
+        {
+            _flywheelIO.setUpperVoltage(_flywheelFeedforward.calculate(_upperVelocitySetpoint) + _upperFlywheelFeedback.calculate(_inputs.upperFlywheelVelocity, _upperVelocitySetpoint));
+        }
+
+        if (_lowerVelocitySetpoint != null)
+        {
+            _flywheelIO.setLowerVoltage(_flywheelFeedforward.calculate(_lowerVelocitySetpoint) + _lowerFlywheelFeedback.calculate(_inputs.lowerFlywheelVelocity, _lowerVelocitySetpoint));
+        }
+    }
+
+    public void setUpperVelocity(double upperVelocitySetpoint)
+    {
+        _upperVelocitySetpoint = MathUtil.clamp(upperVelocitySetpoint, 0, _maxFlywheelSpeed);
```

Comment:
We'll need to forward the velocity onto the IO instance:

```suggestion
        _flywheelIO.setUpperVelocity(MathUtil.clamp(upperVelocitySetpoint, 0, _maxFlywheelSpeed));
```